### ISSUE TITLE
Optimizer state offload for `low_ram` preview generation

### ISSUE THREAD

**Issue by azrahello** [filipstrand/mflux#350]
### Problem

When training with `low_ram: true` at higher resolutions (e.g. 1024×1024) with many LoRA layers (270 layers, rank 32), preview image generation causes heavy swap usage (40GB+ on a 64GB system). This is because the optimizer state (AdamW moments m and v) remains in memory while the preview generation needs to run a full inference pass through the transformer + VAE decode.

### Solution

When `low_ram: true` is set, offload the optimizer state to a temporary file on disk before generating previews, then restore it after:

```python
@staticmethod
def _generate_previews_with_offload(adapter, training_spec, training_state):
    optimizer = training_state.optimizer

    with tempfile.TemporaryDirectory() as tmp_dir:
        # 1. Save optimizer state to disk
        offload_path = Path(tmp_dir) / "optimizer_offload.safetensors"
        optimizer.save(offload_path)

        # 2. Release from memory
        optimizer.optimizer.state = []
        gc.collect()
        mx.clear_cache()

        # 3. Generate previews with freed memory
        TrainingTrainer._generate_previews(adapter, training_spec, training_state)

        # 4. Restore optimizer state
        restored_state = tree_unflatten(list(mx.load(str(offload_path)).items()))
        optimizer.optimizer.state = restored_state

    gc.collect()
    mx.clear_cache()
```

In the training loop, only the `low_ram` path uses the offload:

```python
if training_state.should_generate_image(training_spec):
    if training_spec.low_ram:
        TrainingTrainer._generate_previews_with_offload(adapter, training_spec, training_state)
    else:
        TrainingTrainer._generate_previews(adapter, training_spec, training_state)
```

### Properties

- **Zero impact on training results** — the optimizer state is saved and restored bit-for-bit via safetensors
- **Negligible I/O overhead** — LoRA optimizer moments are small (hundreds of MB), SSD write/read is milliseconds
- **Only active with `low_ram: true`** — normal training path is untouched
- **Reduces swap pressure significantly** during preview generation at higher resolutions

### Environment

- **Chip**: Apple M2 Ultra (76 GPU cores, 64GB unified memory)
- **MLX**: 0.30.6
- **mflux**: 0.16.1

**Comment by filipstrand** [comment]
Tried this out now and seems to work well. But I also noticed that on main, that preview generation there actually increase over time quite substantially (e.g from 3s per step for 40s per step for me(!)... did not notice this before) so now I'm considering always enabling this since this seem to fix that problem. (My guess is that it is a small price to pay to offload the optimiser state compared to all gradient steps in training anyway). I think I'll merge this to main soon, let me know if it helps for you (I still see big memory spikes on my machine, mactop reporting 35GB -> 177GB when training all z-image layers with 1024, 32 rank LoRA for example)

PR:
https://github.com/filipstrand/mflux/pull/356

**Comment by azrahello** [comment]
> [@azrahello](https://github.com/azrahello) Super thankful that you noticed this! I can definitely see the enormous slowdown in main with preview generation, so I chose to always perform this offload now.

No worries, yesterday I was training at 512, but with the base version, and I was spending more time generating previews than actually training. 9 steps are bearable, but 50 steps at 60 sec/it felt excessive to me :D . I’ll try it tomorrow, thanks to you!
